### PR TITLE
[Phase 1] LPS-174289 Remove Stream for module: portal-search-test-util part 2

### PR DESCRIPTION
**Pull request by julschong**
resubmitted from #3653

### CODE REVIEW

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/DocumentsAssert.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ public static void assertValuesIgnoreRelevance(
 
 	public static void assertValuesIgnoreRelevance(
 		String message, List<Document> documentsList, String fieldName,
-		Stream<?> expectedValues) {
+		Collection<?> expectedValues) {
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/DocumentsAssert.java:94]: @julschong Please try to make these methods consistent, most of the usages are passing `List<String>` and it seems only 1 usage is passing `List<Long>`. Just make the `expectedValues` `List<String>` so you can simplify more of `_sort()`.

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/DocumentsAssert.java [hunk 2]
```diff
@@ -174,21 +174,13 @@ private static String _getMessage(
 			message, "->", StringUtil.merge(objects), "->", values);
 	}
 
-	private static String _sort(Collection<String> collection) {
-		List<String> list = new ArrayList<>(collection);
+	private static String _sort(Collection<?> collection) {
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/DocumentsAssert.java:161]: @julschong Please rename it to better reflect what it does.

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseNestedFieldsSortTestCase.java [hunk 3]
```diff
@@ -59,12 +58,10 @@ protected void addDocumentWithOneDDMField(
 	}
 
 	protected void assertSort(String fieldName) {
-		Stream.of(
-			"C", "B", "A"
-		).forEach(
-			value -> addDocumentWithOneDDMField(
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseNestedFieldsSortTestCase.java:65]: @julschong Please double check if there's still usage of `addDocumentWithOneDDMField`.

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/groupby/BaseGroupByTestCase.java [hunk 4]
```diff
@@ -189,23 +187,22 @@ public void testGroupByDocsSortsSortFieldDesc() throws Exception {
 
 	@Test
 	public void testGroupByDocsStart() throws Exception {
-		Map<String, Integer> map1 = HashMapBuilder.put(
+		Map<String, String> map1 = new HashMap<>();
+
+		Map<String, Integer> map2 = HashMapBuilder.put(
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/groupby/BaseGroupByTestCase.java:192]: @julschong Why swapping map1 and map2?

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/groupby/BaseGroupByTestCase.java [hunk 5]
```diff
@@ -189,23 +187,22 @@ public void testGroupByDocsSortsSortFieldDesc() throws Exception {
...
 
-		Stream<Map.Entry<String, Integer>> stream = entries.stream();
-
-		Map<String, String> map2 = stream.collect(
-			Collectors.toMap(
-				Map.Entry::getKey,
-				entry -> getCountPairString(
-					entry.getValue(), entry.getValue() - 1)));
+		for (Map.Entry<String, Integer> entry : map2.entrySet()) {
+			map1.compute(
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/groupby/BaseGroupByTestCase.java:201]: @julschong map.put should enough here.

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/groupby/BaseGroupByTestCase.java [hunk 6]
```diff
@@ -352,15 +348,13 @@ public void testGroupByTermsSizeMoreThanDefault() throws Exception {
...
-		Stream<Map.Entry<String, Integer>> stream = entries.stream();
+		Map<String, String> map2 = new HashMap<>();
 
-		Map<String, String> map2 = stream.collect(
-			Collectors.toMap(
-				Map.Entry::getKey,
-				entry -> getCountPairString(
-					entry.getValue(), entry.getValue())));
+		for (Map.Entry<String, Integer> entry : map1.entrySet()) {
+			map2.computeIfAbsent(
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/groupby/BaseGroupByTestCase.java:354]: @julschong No need to computeIfAbsent here.

Your source is the entry set of another map, the keys won't have duplicate.

**Review by dantewang — CHANGES_REQUESTED** [review]
@julschong Please see my comment, double check all your changes, and resend, thanks.

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/groupby/BaseGroupByTestCase.java (hunk 4 above)
Comment (julschong) [reply modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/groupby/BaseGroupByTestCase.java:192]: @dantewang I'm trying to keep map2 declaration and usage closer together

Comment (julschong) [reply modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/groupby/BaseGroupByTestCase.java:192]: maybe I'm overthinking this

**Pull request closed without merging**